lib/HM/Harmony_Search.hpp

```diff
@@ -0,0 +1,206 @@
+
+#ifndef LIB_HARMONY_SEARCH_HPP_
+#define LIB_HARMONY_SEARCH_HPP_
+
+#define _USE_MATH_DEFINES
+
+#include "./filter_param.hpp"
+#include "./opt_helper.hpp"
+
+#include <stdlib.h>
+#include <math.h>
+#include <stdio.h>
+#include <iostream>
+#include <fstream>
+#include <complex>
+#include <vector>
+#include <time.h>
+#include <random>
+struct Harmony_Search_Parameter
+{
+    protected:
+
+    //ハーモニーサーチのパラメータ
+    int Time_Max;   //更新回数
+    int Harmony_Size; //ハーモニーメモリのサイズ
+    double Bandwidth; //バンド幅
+    double R_a;       //選択比率(大きければ大きい程、ハーモニーメモリ内が使われる)
+    double R_p;       //値調整比率(大きければ大きい程、値を調整する確率が上がる)
+
+    public:
+
+    Harmony_Search_Parameter()
+    :Time_Max(9999990),Harmony_Size(10),Bandwidth(0.1),
+     R_a(0.9),R_p(0.1)
+    {}
+    
+    int get_Time_Max() const
+    { return Time_Max; }
+    int get_Harmony_Size() const
+    { return Harmony_Size; }
+    double get_Bandwidth() const
+    { return Bandwidth; }
+    double get_R_a() const
+    { return R_a; }
+    double get_R_p() const
+    { return R_p; }
+};
+
+struct Harmony
+{
+    protected:
+    vector<double> Harmony;
+
+    public:
+    vector<double> get_Harmony() const
+    { return Harmony; }
+};
+
+struct Harmony_Search_Strategy
+{
+    public:
+    Harmony_Search_Strategy(FilterParam input)
+    :fparam(input)
+    {}
+
+    /*  
+    /   ハーモニーサーチのパラメータ
+    /
+    /   更新回数
+    /   ハーモニーメモリのサイズ
+    /   バンド幅
+    /   選択比率(大きければ大きい程、ハーモニーメモリ内が使われる)
+    /   値調整比率(大きければ大きい程、値を調整する確率が上がる)
+    */
+    Harmony_Search_Parameter Hparam;
+
+    Result optimize();
+    FilterParam fparam;
```

Comment:
これは`HarmonySerachOptimizer`に持たせたいやーつ
```suggestion
    Harmony_Search_Parameter Hparam;

    Result optimize();
```